### DISCUSSION TITLE
Input Object Types with UsePaging Annotation

### DISCUSSION THREAD

**Discussion by rpf3** [ChilliCream/hotchocolate#5501] [Q&A]
When you add the `UsePaging` attribute to a query method, the generated schema automatically gets four arguments: first, last, after, before.  I am in the process of adding filtering/sorting logic into my queries and was looking into creating a single query input type to model both pagination and filtering arguments.  Is it possible to disable this automated input schema generation?  Am I barking up the wrong tree here entirely?

**Comment by PascalSenn** [comment]
@rpf3  Generally it is best practice to have there as separate arguments. Client tooling, like relay, expects these arguments to be there. You could rewrite it (like we did with mutation conventions) but it's probably not worth the effort. What is the motivation behind this?
